weblate/machinery/tests.py

```diff
@@ -532,6 +537,38 @@ def test_google_invalid(self):
         with self.assertRaises(MachineTranslationError):
             self.assert_translate(machine, empty=True)
 
+    def test_google_apiv3_bad_config(self):
+        with self.assertRaisesRegex(
+            MachineTranslationError, r"API\sskey|Cloud\sproject"
+        ):
+            # flake8: noqa: F841
+            machine = self.get_machine(GoogleTranslationV3)
+
+    @override_settings(
+        MT_GOOGLE_CREDENTIALS="SECRET", MT_GOOGLE_PROJECT="translating-7586"
+    )
+    @patch.object(
+        GoogleTranslationV3, "download_languages", Mock(return_value=["cs", "en", "es"])
+    )
+    @patch.object(
+        GoogleTranslationV3,
+        "download_translations",
+        Mock(
+            return_value=[
+                {
+                    "text": "Ahoj",
+                    "quality": 90,
+                    "service": "Google Translate API v3",
+                    "source": "Hello",
+                }
+            ]
+        ),
+    )
```

Comment:
I think that patching google-cloud-translate would be more useful. In this case do simply skip testing download_languages and download_translations methods, what is quite big amount of untested code in the class.

Patching `TranslationServiceClient.translate_text` and `TranslationServiceClient.get_supported_languages` seems like a more useful approach.

Reply:
Can't agree more. Make much more sense. Thanks for  pointing that out.